src/xbot/common/properties/Property.java

```diff
@@ -22,17 +22,42 @@
 
     ITableProxy permanentStore;
     ITableProxy randomAccessStore;
-
+    
+    /** 
+     * New enum to determine property persistence
+     */
+    public enum PropertyPersistenceType{
```

Comment:
Should probably add Javadoc comments to each enum value to explain what they mean.